tests/unit/plugins/test_publisher.py

```diff
@@ -15,22 +15,409 @@
+    publisher_instance, event_message, changes_url,
+        initial_changes, initial_changes_req,
+        initial_changes_pending_json_resp, mock_sleep, caplog):
+    """Ensure exception raised when publish wait timeout exceeded."""
+    publisher_instance.publish_wait_timeout = 0.0001
+    event_message.data['resourceRecords'] = initial_changes_req['additions']
+    publisher_instance.http_client._request_post_mock.side_effect = [
+        initial_changes_pending_json_resp]
+    initial_changes['status'] = 'pending'
+    publisher_instance.http_client._get_json_mock.side_effect = [
+        initial_changes]
+
+    with pytest.raises(exceptions.GCPPublishRecordTimeoutError) as error:
+        await publisher_instance.publish_changes(event_message)
+
+    assert error.match('Timed out while waiting for DNS changes to transition '
+                       'to \'done\' status.')
+    expected_change_id = json.loads(initial_changes_pending_json_resp)['id']
+    publisher_instance.http_client._get_json_mock.assert_called_with(
+        f'{changes_url}/{expected_change_id}')
+    mock_sleep.assert_called_with(1)
+    assert 2 == len(caplog.records)
+
+
+http_exceptions = [
+    ('404', exceptions.GCPHTTPNotFoundError),
+    ('409', exceptions.GCPHTTPConflictError),
+    ('500', exceptions.GCPHTTPError),
+    ('no_code', exceptions.GCPHTTPError)
+]
+
+
+@pytest.mark.parametrize('status_code,http_exception', http_exceptions)
+@pytest.mark.asyncio
+async def test__publish_changes_http_exceptions_raised(
+        publisher_instance, initial_changes_req, status_code,
+        http_exception):
+    """Exception is raised when getting HTTP error from Google API."""
+
+    publisher_instance.http_client._request_post_mock.side_effect = (
+        exceptions.GCPHTTPError(status_code))
+
+    with pytest.raises(http_exception):
+        await publisher_instance._publish_changes(initial_changes_req)
+
+
+@pytest.mark.asyncio
+async def test__publish_changes_returns_change_id(
+        publisher_instance, initial_changes_req,
+        initial_changes_pending_json_resp):
+    """Ensure change ID is returned from Google API."""
+    expected_change_id = json.loads(initial_changes_pending_json_resp)['id']
+    publisher_instance.http_client._request_post_mock.return_value = (
+        initial_changes_pending_json_resp)
+
+    change_id = await publisher_instance._publish_changes(
+        initial_changes_req)
+
+    assert expected_change_id == change_id
+
+
+conflict_types = [
+    (all_existing_zone_records(), handled_conflict_changes_req()),
+    (all_existing_zone_records_empty(), initial_changes_req())
+]
+
+
+@pytest.mark.parametrize('existing,output', conflict_types)
+@pytest.mark.asyncio
+async def test__handle_additions_conflict(
+        publisher_instance, existing, initial_changes_req, output):
+    """Test correctly handling an additions conflict."""
+    publisher_instance.http_client._get_all_mock.return_value = (existing)
```

Comment:
Why the ()?  I don't think they actually do anything here.  If you wanted to create a tuple, you need a comma.